src/canvas-driver.js

```diff
@@ -117,6 +117,34 @@ function translateLine (element, origin) {
   return operations;
 }
 
+function translatePolygon (element, origin) {
+  const [first, ...rest] = element.points;
+
+  return [].concat(
+    [{call: 'beginPath', args: []}],
+    [{call: 'moveTo', args: [origin.x + first.x, origin.y + first. y]}],
+    rest.map(point => {
+      return {call: 'lineTo', args: [origin.x + point.x, origin.y + point.y]};
+    }),
+    [{call: 'closePath', args: []}],
+    element.draw.map(operation => {
+      const fillInstructions = [
+        {set: 'fillStyle', value: operation.fill},
+        {call: 'fill', args: []}
+      ]
+      const strokeInstructions = [
+        {set: 'strokeStyle', value: operation.stroke},
+        {call: 'stroke', args: []}
+      ]
+      return operation.fill
+        ? fillInstructions
+        : operation.stroke
+          ? strokeInstructions
+          : [];
+    })
+  );
```

Comment:
Nice improvement. I wonder if we could go one step further:

```js
return [
  {call: 'beginPath', args: []},
  {call: 'moveTo', args: [origin.x + first.x, origin.y + first. y]},

  rest.map(point => {
    return {call: 'lineTo', args: [origin.x + point.x, origin.y + point.y]};
  }),

  {call: 'closePath', args: []},

  element.draw.map(operation => {
    const fillInstructions = [
      {set: 'fillStyle', value: operation.fill},
      {call: 'fill', args: []}
    ]
    const strokeInstructions = [
      {set: 'strokeStyle', value: operation.stroke},
      {call: 'stroke', args: []}
    ]

    if (operation.fill) {
      return fillInstructions;
    } else if (operation.stroke) {
      return strokeInstructions;
    } else {
      throw new Error('Operation must have stroke or fill property');
    }
  })
];
```

This should be a little more performant because it creates less arrays each frame. I also prefer the `if/else` style to the ternary style, just because it's a little easier to read.

What do you think?